webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala

```diff
@@ -0,0 +1,58 @@
+package com.normation.rudder.services.healthcheck
+
+import java.lang.Runtime.getRuntime
+
+import com.normation.errors.IOResult
+import com.normation.rudder.domain.logger.{HealthcheckLoggerPure => logger}
+import zio.UIO
+import zio.ZIO
+
+sealed trait Status
+case object Ok extends Status
+case object Warning extends Status
+case object Critical extends Status
+
+trait Check {
+  def name: String
+  def run: IOResult[HealthcheckResult]
+}
+
+final case class HealthcheckResult(
+    name  : String
+  , msg   : String
+  , status: Status
+)
```

Comment:
Is there an advantage of that over:
```
sealed trait HealthcheckResult { def name; def msg }
object HealthcheckResult {
  final case class OK(val name: String, val msg: String) extends HealthcheckResult
  final case class Warning(val name: String, val msg: String) extends HealthcheckResult
  final case class Critical(val name: String, val msg: String) extends HealthcheckResult
}
``` ?

Reply:
I think we may put the Check object inside the CheckResult so we could use it's name parameter direclty without errors

Reply:
`Check` is not a structure, it's a service with a name. It can't be passed around in structure, that's forbiden. 
We can add a `CheckName(name: String)`, have `object CheckCoreNumber extends Check(CheckName("CPU Cores available"))` and pass that name object around

Reply:
Not sure to understand the utility of `CheckName` just replace the `name: String` in `Check` ?